src/hevm/src/EVM/SymExec.hs

```diff
@@ -347,41 +351,68 @@ simplify e = if (mapExpr go e == e)
     go (CopySlice (Lit 0x0) (Lit 0x0) (Lit 0x0) _ dst) = dst
 
     -- simplify buffers
+    go o@(ReadWord (Lit a) (WriteWord (Lit b) v _))
+      | a == b = v
+      | otherwise = o
     go o@(ReadWord (Lit _) _) = Expr.simplifyReads o
     go o@(ReadByte (Lit _) _) = Expr.simplifyReads o
 
-    -- redundant Eq
+    -- function selector checks
+    go (SHR (Lit 0xe0) (ReadWord (Lit 0x0) (WriteByte (Lit 0x0) (LitByte sel0) (WriteByte (Lit 0x1) (LitByte sel1) (WriteByte (Lit 0x2) (LitByte sel2) (WriteByte (Lit 0x3) (LitByte sel3) _))))))
+      = Lit $ word $ padLeft 32 $ BS.singleton sel0 <> BS.singleton sel1 <> BS.singleton sel2 <> BS.singleton sel3
+
+    -- concrete LT / GT
+    go (EVM.Types.LT (Lit a) (Lit b))
+      | a < b = Lit 1
+      | otherwise = Lit 0
+    go (EVM.Types.GT (Lit a) (Lit b))
+      | a > b = Lit 1
+      | otherwise = Lit 0
+
+    -- syntactic Eq reduction
+    go (Eq (Lit a) (Lit b))
```

Comment:
I would think/hope this is redundant given the one below? But maybe there is a reason why it's needed, sorry, it's a bit unclear to me. I thought the stuff below should catch this?

Reply:
So the one below applies to all `Eq` expressions, and in that case we don't do any simplifications in the case where `a` is not the exact same expression as `b` (since doing so would require some smt calls or custom decision procedure). 

The case here applies to `Lit`s only and then we can simplify unequal `Lit`s down to `Lit 0`.

Reply:
Ahhh, it's for `Lit 0`. I see, sorry.